packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeEthForm/Inputs.tsx

```diff
@@ -71,42 +81,108 @@ export const Inputs = () => {
 
+    const isFractionButtonDisabled = (divisor: number) => {
+        if (!account.formattedBalance || !network.decimals) return false;
+
+        return new BigNumber(account.formattedBalance)
+            .dividedBy(divisor)
+            .decimalPlaces(network.decimals)
+            .lte(MIN_ETH_AMOUNT_FOR_STAKING);
+    };
+
+    const tooltip = (
+        <Translation
+            id="TR_STAKE_MIN_AMOUNT_TOOLTIP"
+            values={{
+                amount: MIN_ETH_AMOUNT_FOR_STAKING.toString(),
+                symbol: account.symbol.toUpperCase(),
+            }}
+        />
+    );
+
+    const isBalanceBelowMinStake = new BigNumber(account.formattedBalance || '0').lt(
+        MIN_ETH_AMOUNT_FOR_STAKING,
+    );
+
     return (
-        <Column gap={spacings.sm} alignItems="center">
-            <NumberInput
-                name={CRYPTO_INPUT}
-                labelLeft={
-                    <FormFractionButtons
-                        setRatioAmount={setRatioAmount}
-                        setMax={setMax}
-                        symbol={account.symbol}
-                        totalAmount={account.formattedBalance}
-                        decimals={network.decimals}
+        <Column gap={spacings.sm}>
+            <InputWithOptions
+                input={
+                    <NumberInput
+                        name={CRYPTO_INPUT}
+                        locale={locale}
+                        labelLeft={<Translation id="AMOUNT" />}
+                        labelRight={
+                            (isDirty || hasValues) && (
+                                <Button
+                                    type="button"
+                                    variant="tertiary"
+                                    size="tiny"
+                                    onClick={clearForm}
+                                >
+                                    <Translation id="TR_CLEAR_ALL" />
+                                </Button>
+                            )
+                        }
+                        control={control}
+                        rules={cryptoInputRules}
+                        maxLength={formInputsMaxLength.amount}
+                        innerAddon={<Text variant="tertiary">{account.symbol.toUpperCase()}</Text>}
+                        bottomText={errors[CRYPTO_INPUT]?.message ?? null}
+                        inputState={getInputState(cryptoError || fiatError)}
+                        onChange={value => {
+                            onCryptoAmountChange(value);
+                        }}
                     />
                 }
-                labelRight={
-                    (isDirty || hasValues) && (
-                        <Button type="button" variant="tertiary" size="tiny" onClick={clearForm}>
-                            <Translation id="TR_CLEAR_ALL" />
-                        </Button>
-                    )
+                options={[
```

Comment:
I don't know if there's that much benefit in having different API for each of the props, I'd just make this ReactNodes, too.